Add long read benchmark

Itolstoganov opened the pull request:
- Adds [pbsim3 ](https://github.com/yukiteruono/pbsim3/tree/master) simulated Pacbio and Nanopore datasets
- Replaces slurm profile with https://github.com/Snakemake-Profiles/slurm
- Chunks rye genome into 1Gbp fragments (pbsim3 requirement)

Itolstoganov committed:
Simulate RSII and ONT long reads

Itolstoganov committed:
Accuracy estimation for sim1 long read datasets

Itolstoganov committed:
Move ont and clr to separate datasets to enable plots

Itolstoganov committed:
Use suitable minimap presets

Itolstoganov committed:
Comply with mapping-only mode

Itolstoganov committed:
Output incorrect reads with qs > 15

Itolstoganov committed:
Change false hq reads output format

Itolstoganov committed:
Use saeval instead of mapeval to measure accuracy

Itolstoganov committed:
Add HiFi dataset

Itolstoganov committed:
Use hifi preset for minimap

Itolstoganov committed:
Do not downsample hifi datasets

Itolstoganov committed:
Compress .maf files

Itolstoganov committed:
Remove sim1illumina

Itolstoganov committed:
Rename long read datasets, restore short read benchmark

Itolstoganov committed:
Rename sim1 datasets

Itolstoganov committed:
Refactor MAF parsing and long read benchmark rules

Itolstoganov committed:
Remove test section

marcelm commented:
I’m looking into this now (finally, sorry for the delay). I think the SLURM profile change should be left out as it uses an older method of configuring SLURM. But I can remove that commit myself. Do you remember what the problem was?

Itolstoganov commented:
Sure, I agree with that. Iirc the slurm issue was that the default(?) environment variable for the number of cpu per task that is used on dardel is different from the one specified in sbatch.

marcelm commented on simx/Snakefile:
```diff
@@ -105,7 +110,15 @@ def minimap2_profile(wildcards):
     approximately read length 500
     """
-    return "sr" if int(wildcards.read_length) <= 500 else "lr:hq"
+    if (int(wildcards.read_length) <= 500):
```
use a match statement

marcelm commented on Snakefile:
```diff
@@ -272,6 +313,106 @@ rule sim01_long:
+    
+    reference_path = "genomes/" + wildcards.genome + ".fa"
+    ref_len = 0
+    with open(reference_path, "r") as ref:
```
I modified the code to read the much smaller `.fai` file instead because running `snakemake --dry-run` otherwise took something like two minutes because it would repeatedly read and parse the references.

marcelm commented on Snakefile:
```diff
@@ -272,6 +313,106 @@ rule sim01_long:
+                ref_len += len(line.strip())    
+    num_reads = N_READS[mean_read_length]
+    depth = float(num_reads * mean_read_length) / float(ref_len)
+    result += " --depth {}".format(depth)
```
I converted this to f-strings

Itolstoganov commented:
Specifically, the error message for the rule with default resources looks like

`SLURM_CPUS_PER_TASK=3 != SLURM_TRES_PER_TASK=cpu=1`

This occurs when the number of threads determined from `mem_mb` parameter (`mem_mb` divided by `DefMemPerCPU` in partition settings) not the same as the one from `cpus_per_task` .

marcelm commented:
Because splitting of the rye genome into 1 Gbp chunks means that everything on rye has to be re-run, I have merged that change separately, see 06462ba1. (Also, I rewrote it to use `dnaio` which is a faster and IMO easier to use FASTA/FASTQ parsing and writing library that I am co-author of.)

marcelm committed:
Remove unused sim0p1 dataset

marcelm committed:
Comment out unused sim3, sim5 datasets

marcelm committed:
Do not read full FASTA when just evaluating rules

read the .fai instead

in pbsim_parameters

marcelm committed:
Simplify by getting rid of pbsim_outprefix

And remove unused first_bam_name

marcelm committed:
Simplify

marcelm committed:
Download pbsim data files from a release for reproducibility

marcelm committed:
Remove unused parameter

marcelm committed:
Unset TMPDIR for ccs

Forces temporary files to be created in the output directory. Otherwise,
it appars files from different concurrent ccs runs could overwrite each
other.

marcelm committed:
Pin more dependencies

Itolstoganov committed:
Increase file descriptor limit to avoid ccs crash

marcelm committed:
HiFi configuration that reproduces paper plots

marcelm reviewed (approved):
I made a couple of changes (see comments). I also removed the SLURM profile.

Also, I added a `hifi.yaml` file for reproducing the plots from the paper. Everything works after you added ulimit.

I will merge soon, just want to have one last look.
